examples/coco/.tensorleap/dataset.py

```diff
@@ -100,44 +106,38 @@ def input_image(idx, data):
     if len(img.shape) == 2:
         # grey scale -> expand to rgb
         img = gray2rgb(img)
-    img = cv2.resize(img, (image_size, image_size), interpolation=cv2.INTER_LINEAR)
-
+    img = cv2.resize(img, (IMAGE_SIZE, IMAGE_SIZE), interpolation=cv2.INTER_LINEAR)
     # rescale
     img = img / 255
     return img.astype(np.float)
 
 
-def ground_truth_mask(idx, data):
-    print("GT mask")
+# Ground Truth Encoder
+def ground_truth_mask(idx: int, data: SubsetResponse) -> ndarray:
     data = data.data
-    catIds = data['cocofile'].getCatIds(catNms=categories)
+    catIds = data['cocofile'].getCatIds(catNms=CATEGORIES)
     x = data['samples'][idx]
-    batch_masks = []
     annIds = data['cocofile'].getAnnIds(imgIds=x['id'], catIds=catIds, iscrowd=None)
     anns = data['cocofile'].loadAnns(annIds)
     mask = np.zeros([x['height'], x['width']])
     for ann in anns:
         _mask = data['cocofile'].annToMask(ann)
         mask[_mask > 0] = _mask[_mask > 0] * (catIds.index(ann['category_id']) + 1)
-    mask = cv2.resize(mask, (image_size, image_size), interpolation=cv2.INTER_NEAREST)[..., np.newaxis]
+    # here we add other vehicles (truck, bus, train) to the car mask to create a vehicle mask
+    if SUPERCATEGORY_GROUNDTRUTH:
+        car_id = catIds[-1]
+        other_anns_ids = data['cocofile'].getAnnIds(imgIds=x['id'], catIds=data['supercategory_ids'], iscrowd=None)
+        other_anns = data['cocofile'].loadAnns(other_anns_ids)
+        for j, ot_ann in enumerate(other_anns):
+            _mask = data['cocofile'].annToMask(ot_ann)
+            mask[_mask > 0] = _mask[_mask > 0] * (catIds.index(car_id) + 1)
+    mask = cv2.resize(mask, (IMAGE_SIZE, IMAGE_SIZE), interpolation=cv2.INTER_NEAREST)[..., np.newaxis]
     return mask.astype(np.float)
 
 
-def metadata_background_percent(idx, data):
-    print("extracting metadata BG percent")
-    data = data.data
-    catIds = data['cocofile'].getCatIds(catNms=categories)
-    x = data['samples'][idx]
-    batch_masks = []
-    annIds = data['cocofile'].getAnnIds(imgIds=x['id'], catIds=catIds, iscrowd=None)
-    anns = data['cocofile'].loadAnns(annIds)
-    mask = np.zeros([x['height'], x['width']])
-    for ann in anns:
-        _mask = data['cocofile'].annToMask(ann)
-        mask[_mask > 0] = _mask[_mask > 0] * (catIds.index(ann['category_id']) + 1)
-    mask = cv2.resize(mask, (image_size, image_size), interpolation=cv2.INTER_NEAREST)[..., np.newaxis]
-    mask = mask.astype(np.float)
-
+# Metadata functions
+def metadata_background_percent(idx: int, data: SubsetResponse) -> float:
+    mask = ground_truth_mask(idx, data)
     unique, counts = np.unique(mask, return_counts=True)
     unique_per_obj = dict(zip(unique, counts))
     count_obj = unique_per_obj.get(0.0)
```

Comment:
Instead of writing 0.0, use a dictionary for each class to index.
Also replace this duplicate code with a more generic code.